django-stubs/contrib/postgres/fields/ranges.pyi

```diff
@@ -21,32 +25,41 @@ class RangeOperators:
     NOT_GT: Literal["&<"]
     ADJACENT_TO: Literal["-|-"]
 
-class RangeField(models.Field):
+_Range = TypeVar("_Range", bound=Range[Any])
+
+class RangeField(models.Field[Any, _Range]):
     empty_strings_allowed: bool
-    base_field: models.Field
-    range_type: type[Range]
+    base_field: type[models.Field]
```

Comment:
@sobolevn @sudosubin this is still an instance though isn't it?

https://github.com/django/django/blob/main/django/contrib/postgres/fields/ranges.py#L65